pygen/pygen_src/test/riscv_instr_cov_test.py

```diff
@@ -125,26 +127,134 @@ def is_unaligned_mem_access(self):
             return True
         return False
 
-    def get_imm_sign(self):
-        pass
+    def get_imm_sign(self, imm):
+        #TODO: change imm to vsc.int_t(32)
+        out = BitArray(int=imm.val, length=rcs.XLEN)
+        if out[0]:
+            return operand_sign_e["NEGATIVE"]
+        else:
+            return operand_sign_e["POSITIVE"]
 
-    def get_operand_special_value(self):
-        pass
+    def get_div_result(self):
+        #TODO: change rs2_value to vsc.int_t(32)
+        if self.rs2_value.val == 0:
+            return div_result_e["DIV_BY_ZERO"]
+        elif self.rs2_value.val == 1 and self.rs1_value.val == (1 << (rcs.XLEN-1)):
+            return div_result_e["DIV_OVERFLOW"]
+        else:
+            return div_result_e["DIV_NORMAL"]
 
-    def get_imm_special_val(self):
-        pass
+    def get_operand_special_value(self, operand):
+        if operand.val == 0:
+            return special_val_e["ZERO_VAL"]
+        elif operand.val == 1 << (rcs.XLEN-1):
+            return special_val_e["MIN_VAL"]
+        elif operand.val == 1 >> 1:
+            return special_val_e["MAX_VAL"]
+        else:
+            return special_val_e["NORMAL_VAL"]
 
-    def get_compare_result(self):
-        pass
+    def get_imm_special_val(self, imm):
+        if imm.val == 0:
+            return special_val_e["ZERO_VAL"]
+        elif self.format == riscv_instr_format_t.U_FORMAT:
+            # unsigned immediate value
+            # TODO: self.imm_len
+            max = vsc.int_t(32, (1 << self.imm_len)-1)
+            if imm.val == 0:
+                return special_val_e["MIN_VAL"]
+            if imm.val == max.val:
+                return special_val_e["MAX_VAL"]
+        else:
+            # signed immediate value
+            max = vsc.int_t(32, (2 ** (self.imm_len - 1)) - 1)
+            min = vsc.int_t(32, -2 ** (self.imm_len - 1))
+            if min.val == imm.val:
+                return special_val_e["MIN_VAL"]
+            if max.val == imm.val:
+                return special_val_e["MAX_VAL"]
+        return special_val_e["NORMAL_VAL"]
 
-    def get_logical_similarity(self):
-        pass
+    def get_compare_result(self):
+        val1 = vsc.int_t(rcs.XLEN)
+        val2 = vsc.int_t(rcs.XLEN)
+        val1.val = self.rs1_value.val
+        val2.val = self.imm.val if (
```

Comment:
this is a pretty nitty comment, but  why not instantiate these as `val1 = vsc.int_t(rcs.XLEN, self.rs1_value.val)` to match what you do in other functions?

Reply:
Fixed 👍